pennylane/hf/tapering.py

```diff
@@ -451,16 +461,157 @@ def transform_hamiltonian(h, generators, paulix_ops, paulix_sector):
+            )
         )
 
     c = anp.multiply(val, h.terms[0])
     c = qml.math.stack(c)
 
     return _simplify(qml.Hamiltonian(c, o))
+
+
+def optimal_sector(qubit_op, generators, active_electrons):
+    r"""Get the optimal sector which contains the ground state.
+
+    To obtain the optimal sector, we need to choose the right eigenvalue for the symmetries :math:`\vec{\tau}`. We can do
+    so by using the following relation between the Pauli-Z qubit operator and the occupation number under a Jordan-Wigner
+    transform.
+
+    .. math::
+
+        \sigma_{i}^{z} = I - 2a_{i}^{\dagger}a_{i}
+
+    According to this relation, an occupied and unoccupied fermionic mode corresponds to the -1 and +1 eigenvalue of
+    the Pauli-Z operator, respectively. Therefore, to get the optimal sector, for each symmetery :math:`\tau`, we first
+    define a symmetry sector by the wires it acts upon and then calculate the number of occupied orbitals (:math:`N_{OC}`)
+    present in it. If :math:`N_{OC}` is even (odd), then the correct sector for the given symmetry is
+    the one with a +1 (-1) eigenvalue.
+
+    Args:
+        qubit_op (pennylane.Hamiltonian): Hamiltonian for which symmetries are being generated to perform tapering
+        generators (list[pennylane.Hamiltonian]): list of symmetry generators for the Hamiltonian
+        active_electrons (int): The number of active electrons in the system for generating the Hartree-Fock bitstring
+
+    Returns:
+        list[int]: eigenvalues corresponding to the optimal sector which contains the ground state
+
+    .. code-block:: python
+
+        >>> symbols = ['H', 'H']
+        >>> coordinates = np.array([0., 0., -0.66140414, 0., 0., 0.66140414]))
+        >>> mol = qml.hf.Molecule(symbols, coordinates)
+        >>> H, qubits = qml.hf.generate_hamiltonian(mol)(), 4
+        >>> generators, paulix_ops = generate_symmetries(H, qubits)
+        >>> optimal_sector(H, generators, 2)
+          [1, -1, -1]
+
+    """
+
+    if active_electrons < 1:
+        raise ValueError(
+            f"The number of active electrons must be greater than zero;"
+            f"got 'electrons'={active_electrons}"
+        )
+
+    num_orbitals = len(qubit_op.wires)
+
+    if active_electrons > num_orbitals:
+        raise ValueError(
+            f"Number of active orbitals cannot be smaller than number of active electrons;"
+            f" got 'orbitals'={num_orbitals} < 'electrons'={active_electrons}."
+        )
+
+    hf_str = np.where(np.arange(num_orbitals) < active_electrons, 1, 0)
+
+    perm = []
+    for tau in generators:
+        symmstr = np.array([1 if wire in tau.wires else 0 for wire in qubit_op.wires])
+        coeff = -1 if numpy.logical_xor.reduce(numpy.logical_and(symmstr, hf_str)) else 1
+        perm.append(coeff)
+
+    return perm
+
+
+def transform_hartree_fock(generators, paulix_ops, paulix_sector, num_electrons, num_wires):
```

Comment:
Too many underscores! Remember, we want to avoid the code being too `verbose`. Our function naming should look clever, simple, and intuitive.

How about `taper_hf` or `taper_state`? In fact, this makes me wonder if this function could actually work for an arbitrary computational basis state

Reply:
Sorry about that. I have now changed the name to `transform_hf`. Choosing `transfrom` since Soran suggested to have a consistency with the `transform_hamiltonian` function.